File: sources/droid/AdMobWapperApp/admob-wrapper/src/main/java/it/amdev/admob/wrapper/listeners/OnConsentGatheringListener.kt
package it.amdev.admob.wrapper.listeners

interface OnConsentGatheringListener {
    fun onCompleted(canRequestAds: Boolean,
                    privacyOptionsRequired: Boolean)

    fun onCompletedWithError(errorCode: Int,
                             errorMessage: String,
                             canRequestAds: Boolean,
                             privacyOptionsRequired: Boolean)
}

File: sources/apple/ios/AdMobWrapper/Listeners/OnConsentGatheringListener.swift
//
//  OnConsentGatheringListener.swift
//  AdMobWrapper
//

import Foundation

@objc public protocol OnConsentGatheringListener: AnyObject {
    func onCompleted(canRequestAds: Bool,
                     privacyOptionsRequired: Bool)

    func onCompletedWithError(errorCode: Int,
                              errorMessage: String,
                              canRequestAds: Bool,
                              privacyOptionsRequired: Bool)
}
